src/backend/commands/explain_gp.c

```diff
@@ -1473,6 +1473,44 @@ cdbexplain_formatSeg(char *outbuf, int bufsize, int segindex, int nInst)
 }								/* cdbexplain_formatSeg */
 
 
+/*
+ * cdbexplain_formatAgg
+ *	  Convert CdbExplain_Agg to string.
+ *
+ *		outbuf:  		[output] pointer to a char buffer to be filled
+ *		bufsize: 		[input] maximum number of characters to write to outbuf (must be set by the caller)
+ *		CdbExplain_Agg:	[input] pointer to a struct CdbExplain_Agg
+ */
+static void
+cdbexplain_formatAgg(char *outbuf, int bufsize, CdbExplain_Agg agg)
+{
+	Assert(outbuf != NULL && "CDBEXPLAIN: char buffer is null");
+	Assert(bufsize > 0 && "CDBEXPLAIN: size of char buffer is zero");
+	/* check if truncation occurs */
+#ifdef USE_ASSERT_CHECKING
+	int			nchars_written =
+#endif							/* USE_ASSERT_CHECKING */
+	snprintf(outbuf, bufsize, "%10.0f %10.0f %10d %10d", agg.vmax, agg.vsum, agg.vcnt, agg.imax);
+
+	Assert(nchars_written < bufsize &&
+		   "CDBEXPLAIN:  size of char buffer is smaller than the required number of chars");
+}								/* cdbexplain_formatAgg */
```

Comment:
This kind of codes seem not wrong but hack, not sure it's good to make half of assign under a MACRO.
```c
#ifdef USE_ASSERT_CHECKING
	int			nchars_written =
#endif							/* USE_ASSERT_CHECKING */
```


Use **#ifdef USE_ASSERT_CHECKING** and **#else** may duplicate one or two lines, but make it much clear.